tachyon/base/parallelize.h

```diff
@@ -76,6 +88,18 @@ std::vector<ReturnType> ParallelizeMapByChunkSize(Container& container,
   if (chunk_size == 0) return {};
   size_t num_chunks = (std::size(container) + chunk_size - 1) / chunk_size;
   std::vector<ReturnType> values(num_chunks);
+  if (num_chunks == 1) {
+    SpanTy chunk(std::data(container), std::size(container));
+    if constexpr (ArgNum == 1) {
+      values[0] = callback(chunk);
+    } else if constexpr (ArgNum == 2) {
+      values[0] = callback(chunk, 0);
+    } else {
+      static_assert(ArgNum == 3);
+      values[0] = callback(chunk, 0, chunk_size);
+    }
+    return values;
+  }
```

Comment:
Implementing `CallbackReturn` similarly to the above comment can help reduce redundant code.